src/widgets/modals/GenericChooseModal.js

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { FlatList, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
+
+import { PageContentModal } from './PageContentModal';
+
+import { COMPONENT_HEIGHT } from '../../globalStyles';
+
+/**
+ * Generic modal for rendering a full screen modal with a
+ * list of options. On selecting a choice, returns the object
+ * which was selected, the index/row within the data array and
+ * the field which was displayed to the user.
+ * @prop  {array}   data            Array of objects from which a choice must be made from
+ * @prop  {string}  field           The object field which should be displayed to the user
+ * @prop  {string}  title           The title of the modal
+ * @prop  {func}    onPress         Function to call on selection returns {item, index, field}
+ * @prop  {boolean} isOpen          Indicator for if the modal should be open
+ * @prop  {int}     highlightIndex  Index of the data array for which object should be 'highlighted'
+ */
+export default class GenericChooseModal extends React.PureComponent {
+  keyExtractor = (item, index) => {
+    const { field } = this.props;
+    const content = field && item ? item[field] : item;
+    return `${content}${index}`;
+  };
+
+  renderRow = ({ item, index }) => {
+    const { onPress, field, highlightIndex } = this.props;
+    const { row, text } = localStyles;
+
+    const indexToHighlight = highlightIndex >= 0 ? highlightIndex : 0;
```

Comment:
keyToHighlight and indexToHighlight ?
```
let shouldHighlight = false;
if(keyToDisplay && valueToHighlight) shouldHighlight= item[keyToDisplay] === valueToHighlight 
else if (indexToHighlight)  index === indexToHighlight // null !== 0, undefined !== 0
```

Reply:
done 2e47c9604fb499646169a7b3497724fcda948086

Defaults to highlighting the first row if no `highlightIndex` or `highlightValue` is used. 

This will highlight multiple rows if there are rows with the same value. I think that this is fine to leave as the responsibility of the caller as there are other problems with having multiple same values in a drop down